packages/squiggle-lang/__tests__/Distributions/Symbolic_test.res

```diff
@@ -0,0 +1,177 @@
+    ))
+    theMean 
+    -> unpackFloat 
+    -> expect 
+    -> toBeCloseTo(1.0 /. (1.0 +. (beta /. alpha)))  // https://en.wikipedia.org/wiki/Beta_distribution#Mean
+  })
+
+  // TODO: When we have our theory of validators we won't want this to be NaN but to be an error.
+  test("of beta(0, 0)", () => {
+    let theMean = run(FromDist(
+      ToFloat(#Mean), 
+      GenericDist_Types.Symbolic(#Beta({alpha: 0.0, beta: 0.0}))
+    ))
+    theMean
+    -> unpackFloat
+    -> expect
+    -> ExpectJs.toBeFalsy
+  })
+
+  testAll("of lognormal distributions", list{(2.0, 4.0), (1e-7, 1e-2), (-1e6, 10.0), (1e3, -1e2), (-1e8, -1e4), (1e2, 1e-5)}, tup => { 
+    let (mu, sigma) = tup
+    let theMean = run(FromDist(
+      ToFloat(#Mean), 
+      GenericDist_Types.Symbolic(#Lognormal({mu: mu, sigma: sigma}))
+    ))
+    theMean 
+    -> unpackFloat 
+    -> expect 
+    -> toBeCloseTo(Js.Math.exp(mu +. sigma ** 2.0 /. 2.0 ))  // https://brilliant.org/wiki/log-normal-distribution/
+  })
+
+  testAll("of uniform distributions", list{(1e-5, 12.345), (-1e4, 1e4), (-1e16, -1e2), (5.3e3, 9e9)}, tup => {
+    let (low, high) = tup
+    let theMean = run(FromDist(
+      ToFloat(#Mean), 
+      GenericDist_Types.Symbolic(#Uniform({low: low, high: high}))
+    ))
+    theMean 
+    -> unpackFloat 
+    -> expect 
+    -> toBeCloseTo((low +. high) /. 2.0)  // https://en.wikipedia.org/wiki/Continuous_uniform_distribution#Moments
+  })
+
+  test("of a float", () => {
+    let theMean = run(FromDist(
+      ToFloat(#Mean), 
+      GenericDist_Types.Symbolic(#Float(7.7))
+    ))
+    theMean -> unpackFloat -> expect -> toBeCloseTo(7.7)
+  })
+})
+
+describe("Normal distribution with sparklines", () => {
+
+  let parameterWiseAdditionPdf = (n1: SymbolicDistTypes.normal, n2: SymbolicDistTypes.normal) => {
+    let normalDistAtSumMeanConstr = SymbolicDist.Normal.add(n1, n2)
+    let normalDistAtSumMean: SymbolicDistTypes.normal = switch normalDistAtSumMeanConstr {
+      | #Normal(params) => params
+    }
+    x => SymbolicDist.Normal.pdf(x, normalDistAtSumMean)
+  }
+
+  let normalDistAtMean5: SymbolicDistTypes.normal = {mean: 5.0, stdev: 2.0}
+  let normalDistAtMean10: SymbolicDistTypes.normal = {mean: 10.0, stdev: 2.0}
+  let range20Float = E.A.rangeFloat(0, 20) // [0.0,1.0,2.0,3.0,4.0,...19.0,]
+
+  test("mean=5 pdf", () => {
+    let pdfNormalDistAtMean5 = x => SymbolicDist.Normal.pdf(x, normalDistAtMean5)
+    let sparklineMean5 = fnImage(pdfNormalDistAtMean5, range20Float) 
+    Sparklines.create(sparklineMean5, ()) 
+    -> expect 
+    -> toEqual(`▁▂▃▅███▅▃▂▁▁▁▁▁▁▁▁▁▁▁`)
+  })
+  
+  test("parameter-wise addition of two normal distributions", () => {  
+    let sparklineMean15 = normalDistAtMean5 -> parameterWiseAdditionPdf(normalDistAtMean10) -> fnImage(range20Float)
+    Sparklines.create(sparklineMean15, ())
+    -> expect
+    -> toEqual(`▁▁▁▁▁▁▁▁▁▁▂▃▅▇███▇▅▃▂`)
+  })
+
+  test("mean=5 cdf", () => {
```

Comment:
mean=10 cdf?

Reply:
good catch

Reply:
it's changed in my local, will be in the next push.